src/posthog-surveys.ts

```diff
@@ -99,6 +99,26 @@ export class PostHogSurveys {
                 const targetingFlagCheck = survey.targeting_flag_key
                     ? this.instance.featureFlags.isFeatureEnabled(survey.targeting_flag_key)
                     : true
+
+                let internalTargetingFlagCheck = undefined
+                if (survey.internal_targeting_flag_key) {
+                    // because this feature flag depends on user properties that might not be available yet,
+                    // the decide API will not return this flag when asked for it.
+                    // as a result, we need to distinguish between undefined and false.
+                    // where undefined means that the properties are not yet available.
+                    const value = this.instance.featureFlags.getFeatureFlag(survey.internal_targeting_flag_key)
+                    if (value == undefined) {
+                        internalTargetingFlagCheck = true
+                    } else {
+                        internalTargetingFlagCheck = this.instance.featureFlags.isFeatureEnabled(
+                            survey.internal_targeting_flag_key
+                        )
+                    }
+                } else {
+                    internalTargetingFlagCheck = true
+                }
```

Comment:
Might be unnecessary, but the functional programmer is me says that this evaluation process could be a function instead of mutating the value of a variable.  e.g., the logic (as written) could be rewritten thusly (using js bc I don't have type hints in github, sorry)
```js
const getInternalTargetingFlagCheck = (survey, instance) => {
    if (!survey.internal_targeting_flag_key) {
        return true;
    }

    const flagValue = instance.featureFlags.getFeatureFlag(survey.internal_targeting_flag_key);

    if (flagValue == undefined) {
        return true;
    }

    return instance.featureFlags.isFeatureEnabled(survey.internal_targeting_flag_key); // returns true, false, or undefined
};

const internalTargetingFlagCheck = getInternalTargetingFlagCheck(survey, this.instance); // then use this value
```